python/cugraph/cugraph/experimental/datasets/dataset.py

```diff
@@ -55,39 +62,29 @@ class Dataset:
 
     """
     def __init__(self, meta_data_file_name):
-        self.__dir_path = Path(__file__).parent.absolute()
-
-        self._meta_data_file_name = meta_data_file_name
-        self.__read_meta_data_file(self._meta_data_file_name)
-
-        self._dd = default_download_dir.dd
+        self.__read_meta_data_file(meta_data_file_name)
+        self._dl_path = default_download_dir
         self._edgelist = None
         self._graph = None
         self._path = None
 
     def __read_meta_data_file(self, meta_data_file):
-        metadata_path = self.__dir_path / meta_data_file
+        metadata_path = Path(__file__).parent.absolute() / meta_data_file
         with open(metadata_path, 'r') as file:
             self.metadata = yaml.safe_load(file)
-            file.close()
-
-    # check for updates for download_dir location within obj
-    def __refresh_dd(self):
-        if self._dd != default_download_dir.dd:
-            self._dd = default_download_dir.dd
 
     def __download_csv(self, url):
-        if not os.path.isdir(self._dd):
-            os.makedirs(self._dd)
+        if not os.path.isdir(self._dl_path.path):
+            os.makedirs(self._dl_path.path)
 
         filename = self.metadata['name'] + self.metadata['file_type']
-        if os.path.isdir(self._dd):
+        if os.path.isdir(self._dl_path.path):
```

Comment:
Might be better to use the methods on the `Path` object instead of `os.path` functions when possible:
```suggestion
        if self._dl_path.path.isdir():
```